library/core/src/num/f128.rs

```diff
@@ -454,11 +454,14 @@ impl f128 {
     }
 
     /// Returns `true` if `self` has a positive sign, including `+0.0`, NaNs with
-    /// positive sign bit and positive infinity. Note that IEEE 754 doesn't assign any
-    /// meaning to the sign bit in case of a NaN, and as Rust doesn't guarantee that
-    /// the bit pattern of NaNs are conserved over arithmetic operations, the result of
-    /// `is_sign_positive` on a NaN might produce an unexpected result in some cases.
-    /// See [explanation of NaN as a special value](f128) for more info.
+    /// positive sign bit and positive infinity.
+    ///
+    /// Note that IEEE 754 doesn't assign any meaning to the sign bit in case of
+    /// a NaN, and as Rust doesn't guarantee that the bit pattern of NaNs are
+    /// conserved over arithmetic operations, the result of `is_sign_positive` on
+    /// a NaN might produce an unexpected or non-portable result. See [explanation
+    /// of NaN as a special value](f32) for more info. Use `self >= 0.0` if you
+    /// need fully portable behavior and are okay with `-0.0` being considered positive.
```

Comment:
"Use `self >= 0.0` ... with `-0.0` being considered positive" is this accurate? It seems either the first part should be `self > 0.0` or the second part should be "being considered nonnegative".

Reply:
This function tests whether something is "positive". It considers +0.0 to be positive but -0.0 to not be positive. If you replace the function by `self >= 0.0` to test for positivity, then -0.0 would also be considered positive.

I guess the difference is whether you consider "positive" to mean "strictly positive" or not, but given that `0.0.is_sign_positive()` is true, I think the terminology here is already aligned towards non-strict positivity.